src/OpenTelemetry.Resources.Process/ProcessDetector.cs

```diff
@@ -8,26 +8,49 @@ namespace OpenTelemetry.Resources.Process;
 /// </summary>
 internal sealed class ProcessDetector : IResourceDetector
 {
+    private readonly ProcessDetectorOptions options;
+
+    internal ProcessDetector(ProcessDetectorOptions? options)
+    {
+        this.options = options ?? new ProcessDetectorOptions();
+    }
+
     /// <summary>
     ///     Detects the resource attributes for process.
     /// </summary>
     /// <returns>Resource with key-value pairs of resource attributes.</returns>
     public Resource Detect()
     {
-        return new Resource(new List<KeyValuePair<string, object>>(2)
+        using var currentProcess = System.Diagnostics.Process.GetCurrentProcess();
+        var attributes = new List<KeyValuePair<string, object>>(2)
```

Comment:
The dictionary is having its capacity pre-allocated here, but the number hasn't been updated to account for the new properties.